### ISSUE TITLE
🐛 [BUG] - <IntelliGraph data structure request during batch-run>

### ISSUE THREAD

**Issue by bjoernTWK** [dlr-gtlab/gtlab-core#1305]
### Summary

I tried to run a task within an GTlab-project that runs in the GUI without any problem.
The computation is not started as long as the GTlabIntelliGraph.dll is available. The following errror message occure:
  
  DEBUG [17:03:43] [Batch] Creating temporary batch session '_tmp_batch_session_16528'
  DEBUG [17:03:43] [module data upgrader] Upgrade available for 'IntelliGraph (0.10.4)' to version 0.12.0
  DEBUG [17:03:43] [Batch] process run...
  DEBUG [17:03:43] [Core] Loading project 'DLR-FFE-CoreDrivenFanEngine_final'...
  ERROR [17:03:43] [Core] Project 'DLR-FFE-CoreDrivenFanEngine_final' requires updates to the data structure. Run the project data upgrade command first!
  ERROR [17:03:43] [Batch] could not open project!  ( DLR-FFE-CoreDrivenFanEngine_final )
  DEBUG [17:03:43] [module data upgrader] Upgrade available for 'IntelliGraph (0.10.4)' to version 0.12.0

If the GTlabIntelliGraph.dll is removed, the batch-application carries out the task without any problem.
(I've also tried to add and remove the IntelliGraph within GUI but there was no possibillity to update the data structure)
![image](https://github.com/user-attachments/assets/e8e48359-6a28-459c-929b-7324cc81ac40)

### Expected Behaviour

The run via batch-application lead to the same behavior as in the GUI. Moreover, there has to be a possibility to update the structure (in my opinion a not used Module should not lead to a computation exit).

### Reproduction steps

```bash
1. Load and run a model with an deprecated IntelliGraph-data-structure via batch application (GTlabConsole.exe run -f "FFE-Core-Fan-Engine\project.gtlab" TestTask UserGroup
```

### Screenshots

![DESCRIPTION](LINK.png)

### Logs

```bash
DEBUG [17:03:43] [Core] Loading project 'DLR-FFE-CoreDrivenFanEngine_final'...
ERROR [17:03:43] [Core] Project 'DLR-FFE-CoreDrivenFanEngine_final' requires updates to the data structure. Run the project data upgrade command first!
ERROR [17:03:43] [Batch] could not open project!  ( DLR-FFE-CoreDrivenFanEngine_final )
DEBUG [17:03:43] [module data upgrader] Upgrade available for 'IntelliGraph (0.10.4)' to version 0.12.0
```

### GTlab environment

```bash
GTlab Version: 2.0.10
IntelliGrapgh Version: 0.12.0
```

### OS

Windows

**Comment by jensschmeink** [comment]
@mariusalexander Yes, even modules which are not "standAlone" may change the datamodel in an upgrade. For example a module for compressor design may change its calculator but also tool specific sub elements in compressor datamodel as sub elements of "PreDesign". Currently objects do not know to which module they belong

**Comment by mariusalexander** [comment]
I see, so this is a bit tricky to implement. Theoretically GTlab Core should be able to know which Class originates from which Module, as they need to be registered using the GtDataModelInterface.

**Comment by jensschmeink** [comment]
> I see, so this is a bit tricky to implement. Theoretically GTlab Core should be able to know which Class originates from which Module, as they need to be registered using the GtDataModelInterface.

See also #1171
